packages/react-select/src/Select.js

```diff
@@ -1206,18 +1206,25 @@ export default class Select extends Component<Props, State> {
         this.selectOption(focusedOption);
         break;
       case 'Enter':
+        const targetIsButton = event.target && event.target.type === 'button';
+
+        if (targetIsButton) {
+          return;
+        }
+
         if (event.keyCode === 229) {
           // ignore the keydown event from an Input Method Editor(IME)
           // ref. https://www.w3.org/TR/uievents/#determine-keydown-keyup-keyCode
           break;
         }
+
         if (menuIsOpen) {
-          if (!focusedOption) return;
+          if (!focusedOption) break;
```

Comment:
Since the purpose of break is simply to fall through, why not instead simply call `return event.preventDefault()`